giraffe/src/components/GeoLayer.tsx

```diff
@@ -155,11 +162,11 @@ const onAutoResize = (
 }
 
 const GeoLayer: FunctionComponent<OwnProps> = React.memo(props => {
```

Comment:
This is the main component `GeoLayer`. All the other things above this line doesn't belong in here. We want to keep our components as specific as possible. The functions above this line can go into a different file.

Reply:
I don't agree with this but it is done. 

The GeoLayer function is based on a proxy pattern. Its only purpose is to stop excessive rendering when it is not needed and all the rendering is then delegated to the `Geo` component. What it does can hardly be anymore specific and I doubt that splitting it into two parts makes it more readable.